x-pack/plugin/sql/qa/server/src/main/resources/subselect.sql-spec

```diff
@@ -1,54 +1,68 @@
 // To mute tests follow example in file: example.sql-spec
 
-simpleSubSelect
+basic
 SELECT first_name, last_name FROM (SELECT * FROM test_emp);
-simpleSubSelectWithProjection
+basicProjection
 SELECT first_name, last_name FROM (SELECT first_name, last_name FROM test_emp);
-simpleSubSelectWithAliasedProjection
+basicAliasedProjection
 SELECT last_name, f FROM (SELECT first_name AS f, last_name FROM test_emp);
-simpleSubSelectWithCondition
+basicCondition
 SELECT first_name, last_name FROM (SELECT * FROM test_emp WHERE gender IS NOT NULL) ORDER BY first_name NULLS first;
-simpleSubSelectWithOrderByAlias
+basicOrderByAlias
 SELECT last_name, f FROM (SELECT first_name AS f, last_name FROM test_emp) ORDER BY f DESC NULLS last;
-simpleSubSelectWithOrderByFieldName
+basicOrderByFieldName
 SELECT last_name, f FROM (SELECT first_name AS f, last_name FROM test_emp) ORDER BY last_name ASC;
-simpleSubSelectWithGroupBy
+basicGroupBy
 SELECT gender FROM (SELECT first_name AS f, last_name, gender FROM test_emp) GROUP BY gender ORDER BY gender ASC;
+basicGroupByAlias
+SELECT g FROM (SELECT first_name AS f, last_name, gender AS g FROM test_emp) GROUP BY g ORDER BY g ASC;
+// @AwaitsFix(bugUrl = "follow-up to https://github.com/elastic/elasticsearch/pull/67216")
+basicGroupByWithFilterByAlias-Ignore
+SELECT g FROM (SELECT first_name AS f, last_name, gender AS g FROM test_emp) WHERE g IS NOT NULL GROUP BY g ORDER BY g ASC;
 
-subSelectWithCountAndComplexCondition
+
+countAndComplexCondition
 SELECT COUNT(*) as c FROM (SELECT * FROM test_emp WHERE gender IS NOT NULL) WHERE ABS(salary) > 0 GROUP BY gender ORDER BY gender;
-subSelectWithCountAndGroupBy
+countAndGroupBy
 SELECT COUNT(*), gender FROM (SELECT first_name AS f, last_name, gender FROM test_emp) GROUP BY gender ORDER BY gender ASC;
-subSelectWithCountAndMax
+countAndGroupByByAlias
+SELECT COUNT(*), g FROM (SELECT first_name AS f, last_name, gender AS g FROM test_emp) GROUP BY g ORDER BY g ASC;
+countAndMax
 SELECT COUNT(*), gender, MAX("salary") FROM (SELECT first_name AS f, last_name, gender, salary FROM test_emp) GROUP BY gender ORDER BY gender ASC;
-subSelectWithCountAndMultipleAggregates
+countAndMaxByAlias
+SELECT COUNT(*), g, MAX("s") FROM (SELECT first_name AS f, last_name, gender AS g, salary AS s FROM test_emp) GROUP BY g ORDER BY g ASC;
+// @AwaitsFix(bugUrl = "follow-up to https://github.com/elastic/elasticsearch/pull/67216")
+countAndMaxByAliasAndFilter-Ignore
+SELECT COUNT(*), g, MAX("s") FROM (SELECT first_name AS f, last_name, gender AS g, salary AS s FROM test_emp) WHERE g IS NOT NULL GROUP BY g ORDER BY g ASC;
```

Comment:
This is a new issue discovered when there's an extra plan between the aggregate and its base that prevents the reference attribute to be resolved easily.
And that's because the foldAggregate has a different mechanism around considering aliases which breaks in this case.
I've put the tests in place to have this fixed as a follow-up